components/brave_shields/browser/https_everywhere_service.cc

```diff
@@ -105,14 +105,7 @@ HTTPSEverywhereService::HTTPSEverywhereService(
 }
 
 HTTPSEverywhereService::~HTTPSEverywhereService() {
-  Cleanup();
-}
-
-void HTTPSEverywhereService::Cleanup() {
-  GetTaskRunner()->PostTask(
-      FROM_HERE,
-      base::Bind(&HTTPSEverywhereService::CloseDatabase,
-                 AsWeakPtr()));
+  GetTaskRunner()->DeleteSoon(FROM_HERE, level_db_);
```

Comment:
Q) Is there a reason why not delete db here? Or just delaying time consuming tasks?

Reply:
I assume it must be deleted on the background task runner, not on the UI thread

Reply:
I see!